### ISSUE TITLE
Contrasts to study interactions

### ISSUE THREAD

**Issue by jenellewallace** [GabrielHoffman/dreamlet_analysis#3]
Hi, thanks so much for the awesome package - I was working with edgeR and limma previously, and it seems like dreamlet is exactly the package I need to fully model my experimental design so very happy to be trying it out.

I just have a syntax question - I am trying to use a contrast to study the interaction between two variables. So my model equation is: 
`form ~ 0 + variable1 + variable2 + 1|batch + 1|individual
`
I want to form a contrast as follows:
`a = variable1value1_variable2value1 - variable1value2_variable2value1
`
What is the correct syntax to do this? I tried the syntax above but then when I try to plot the results from that contrast, I get the error:
```
 "Error in topTable(x, coef = coef, number = Inf) : 
  coef must be in coefNames"
```
and I can also see that the coefficient `a` does not exist in `res_dl$celltype$contrasts` (where res_dl is the results of my dreamlet run and celltype is a particular celltype I'm studying). However, I did not get any error message from the dreamlet command. Other contrasts I have made (for example, `b = variable1value1 - variable1value2`) work as expected. I know I can also try including an interaction term in the model form itself, but I wanted to compare the results of models with and without the interaction term. Thanks for any guidance!

**Comment by GabrielHoffman** [comment]
Hi Jenelle, 
Can you give a concrete example we can walk through, ideally with reproducible example.  It's currently too abstract to get an intuition about what you're trying to do

Best,
Gabriel

**Comment by GabrielHoffman** [comment]
Following up here.  If there's a bug, I'd like to fix it

**Comment by jenellewallace** [comment]
Hi Gabriel,
Thanks for following up - there is no bug! In my original question, I realized I just misunderstood how contrasts work and that they need to be formed from existing coefficients in the model (so I realized I needed to create a model with an interaction term in order to have contrasts based on interaction terms). In fact, there was a warning message from dreamlet -> makeContrastsDream: object not found when I tried to use nonexistent coefficients to make contrasts. So I think everything is working as it should!
